node/src/components/linear_chain/state.rs

```diff
@@ -89,11 +111,197 @@ impl LinearChain {
     }
 
     /// Handles registering an upgrade activation point.
-    pub(super) fn got_upgrade_activation_point(&mut self, activation_point: ActivationPoint) {
+    pub(super) fn handle_upgrade_activation_point(&mut self, activation_point: ActivationPoint) {
         debug!(?activation_point, "got an activation point");
         self.next_upgrade_activation_point = Some(activation_point);
     }
 
+    /// Called when linear chain component receives a newly finalized linear chain block.
+    pub(super) fn handle_new_block(
+        &mut self,
+        block: Box<Block>,
+        execution_results: HashMap<DeployHash, ExecutionResult>,
+    ) -> Outcomes {
+        vec![Outcome::StoreBlock(block, execution_results)]
+    }
+
+    /// Called when component gets notified about the fact of the `block` being successfully stored
+    /// in the database.
+    pub(super) fn handle_put_block_result(
+        &mut self,
+        block: Box<Block>,
+        verifiable_chunked_hash_activation: EraId,
+    ) -> Outcomes {
+        let mut outcomes = Vec::new();
+        let pending_signatures = self.new_block(&block);
+        self.update_latest_block(block.clone(), verifiable_chunked_hash_activation);
+        if !pending_signatures.is_empty() {
+            let mut block_signatures = BlockSignatures::new(*block.hash(), block.header().era_id());
+            for sig in pending_signatures.iter() {
+                block_signatures.insert_proof(sig.public_key(), sig.signature());
+            }
+            let should_upgrade = self.should_upgrade(&block_signatures);
+            outcomes.push(Outcome::StoreBlockSignatures(
+                block_signatures,
+                should_upgrade,
+            ));
+            for signature in pending_signatures {
+                if signature.is_local() {
+                    outcomes.push(Outcome::Gossip(Box::new(signature.to_inner().clone())));
+                }
+                outcomes.push(Outcome::AnnounceSignature(signature.take()));
+            }
+        };
+        outcomes.push(Outcome::AnnounceBlock(block));
+        outcomes
+    }
+
+    /// Called upon receiving the result of reading finality signatures from the database.
+    pub(super) fn handle_stored_signatures_result(
```

Comment:
This confused me at first, because I thought it's the result of trying to store signatures.
```suggestion
    pub(super) fn handle_get_stored_signatures_result(
```

Reply:
Done in 0b1ff39c1.